libs/partners/google-genai/langchain_google_genai/chat_models.py

```diff
@@ -0,0 +1,556 @@
+       Must be positive."""
+    n: int = 1
+    """Number of chat completions to generate for each prompt. Note that the API may
+       not return the full n completions if duplicates are generated."""
+
+    _generative_model: Any  #: :meta private:
+
+    @property
+    def lc_secrets(self) -> Dict[str, str]:
+        return {"google_api_key": "GOOGLE_API_KEY"}
+
+    @property
+    def _llm_type(self) -> str:
+        return "chat-google-generative-ai"
+
+    @property
+    def _is_geminiai(self) -> bool:
+        return self.model is not None and "gemini" in self.model
+
+    @classmethod
+    def is_lc_serializable(self) -> bool:
+        return True
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        google_api_key = get_from_dict_or_env(
+            values, "google_api_key", "GOOGLE_API_KEY"
+        )
+        try:
+            import google.generativeai as genai
+
+            genai.configure(api_key=google_api_key)
+        except ImportError:
+            raise ChatGoogleGenerativeAIError(
+                "Could not import google.generativeai python package. "
+                "Please install it with `pip install google-generativeai`"
+            )
+
+        values["client"] = genai
+        if (
+            values.get("temperature") is not None
+            and not 0 <= values["temperature"] <= 1
+        ):
+            raise ValueError("temperature must be in the range [0.0, 1.0]")
+
+        if values.get("top_p") is not None and not 0 <= values["top_p"] <= 1:
+            raise ValueError("top_p must be in the range [0.0, 1.0]")
+
+        if values.get("top_k") is not None and values["top_k"] <= 0:
+            raise ValueError("top_k must be positive")
+        model = values["model"]
+        values["_generative_model"] = genai.GenerativeModel(model_name=model)
+        return values
+
+    @property
+    def _identifying_params(self) -> Dict[str, Any]:
+        """Get the identifying parameters."""
+        return {
+            "model": self.model,
+            "temperature": self.temperature,
+            "top_k": self.top_k,
+            "n": self.n,
+        }
+
+    @property
+    def _generation_method(self) -> Callable:
+        return self._generative_model.generate_content
+
+    @property
+    def _async_generation_method(self) -> Awaitable:
+        # TODO Add support once Google uncomments the async client
+        return self._generative_model.generate_content
+
+    def _prepare_params(
+        self, messages: Sequence[BaseMessage], stop: Optional[List[str]]
+    ) -> Dict[str, Any]:
+        contents = _messages_to_genai_contents(messages)
+        gen_config = {
+            k: v
+            for k, v in {
+                "candidate_count": self.n,
+                "temperature": self.temperature,
+                "stop_sequences": stop,
+                "max_output_tokens": self.max_output_tokens,
```

Comment:
should top_k and top_p be here?

Reply:
Think only these are supported now
```
class GenerationConfig:
    candidate_count: int | None = 0
    stop_sequences: Iterable[str] | None = dataclasses.field(default_factory=list)
    max_output_tokens: int | None = 0
    temperature: float | None = 0.0
    ```

Reply:
![Screenshot 2023-12-13 at 10 39 13 AM](https://github.com/langchain-ai/langchain/assets/22008038/d8b8ed80-84be-494e-9f25-c42da8da3115)

https://ai.google.dev/api/python/google/generativeai/GenerationConfig